runtime/compiler/z/codegen/J9S390PrivateLinkage.cpp

```diff
@@ -1540,18 +1541,20 @@ TR::S390PrivateLinkage::buildVirtualDispatch(TR::Node * callNode, TR::RegisterDe
             }
             break;
          default:
+            // For remote compilations we always need to regenerate the thunk inside the code cache for this compilation.
             if (fej9->needsInvokeExactJ2IThunk(callNode, comp()))
                {
-               comp()->getPersistentInfo()->getInvokeExactJ2IThunkTable()->addThunk(
-                  TR::S390J9CallSnippet::generateInvokeExactJ2IThunk(callNode, sizeOfArguments, methodSymbol->getMethod()->signatureChars(), cg()), fej9);
+               TR_J2IThunk *thunk = TR::S390J9CallSnippet::generateInvokeExactJ2IThunk(callNode, sizeOfArguments, methodSymbol->getMethod()->signatureChars(), cg());
+               fej9->setInvokeExactJ2IThunk(thunk, comp());
                }
             break;
          }
       }
    else
       {
       virtualThunk = fej9->getJ2IThunk(methodSymbol->getMethod(), comp());
-      if (!virtualThunk)
+      // For remote compilations we always need to regenerate the thunk inside the code cache for this compilation.
```

Comment:
On x86 the code is like this:
```
         {
         virtualThunk = fej9->getJ2IThunk(methodSymbol->getMethod(), comp());
         // in server mode, we always need to regenerate the thunk inside the code cache for this compilation
         if (!virtualThunk)
            virtualThunk = fej9->setJ2IThunk(methodSymbol->getMethod(), generateVirtualIndirectThunk(callNode), comp());
         }
```
The check for `comp()->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER` has gone away which makes the code exactly the same as the old one. No change needed!